pkg/lorry/engines/models/replca_role_types.go

```diff
@@ -29,3 +34,20 @@ const (
 	LEARNER   = "Learner"
 	CANDIDATE = "Candidate"
 )
+
+func IsLeaderOrPrimaryOrMaster(role string) bool {
+	return role == LEADER || role == PRIMARY || role == MASTER
```

Comment:
The code attempts to enumerate all roles that appear to be a "primary". However, it is challenging to exhaustively list all possible naming variations here. For example, users might use abbreviations like "L" or "F" when naming roles.

Nevertheless, this issue can be addressed and resolved later, as a TODO.